src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchAPIIntegrationTest.kt

```diff
@@ -0,0 +1,123 @@
+package uk.gov.justice.hmpps.offendersearch.controllers
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.restassured.RestAssured
+import io.restassured.config.ObjectMapperConfig
+import io.restassured.config.RestAssuredConfig
+import org.assertj.core.api.Assertions
+import org.elasticsearch.client.RestHighLevelClient
+import org.hamcrest.CoreMatchers
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.TestContext
+import org.springframework.test.context.TestExecutionListeners
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
+import org.springframework.test.context.support.AbstractTestExecutionListener
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener
+import uk.gov.justice.hmpps.offendersearch.dto.OffenderDetail
+import uk.gov.justice.hmpps.offendersearch.util.LocalStackHelper
+import java.lang.reflect.Type
+import java.util.*
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("dev,localstack")
+@RunWith(SpringJUnit4ClassRunner::class)
+@TestExecutionListeners(listeners = [DependencyInjectionTestExecutionListener::class, OffenderSearchAPIIntegrationTest::class])
+@ContextConfiguration
+class OffenderSearchAPIIntegrationTest : AbstractTestExecutionListener() {
+  @Value("\${test.token.good}")
+  private lateinit var validOauthToken: String
+
+  override fun beforeTestClass(testContext: TestContext) {
+    val objectMapper = testContext.applicationContext.getBean(ObjectMapper::class.java)
+    val esClient = testContext.applicationContext.getBean(RestHighLevelClient::class.java)
+    LocalStackHelper(esClient).loadData()
+    RestAssured.port = Objects.requireNonNull(testContext.applicationContext.environment.getProperty("local.server.port")).toInt()
+    RestAssured.config = RestAssuredConfig.config().objectMapperConfig(
+        ObjectMapperConfig().jackson2ObjectMapperFactory { aClass: Type?, s: String? -> objectMapper })
+  }
+
+  @Test
+  fun surnameSearch() {
+    val results = RestAssured.given()
+        .auth()
+        .oauth2(validOauthToken)
+        .contentType(MediaType.APPLICATION_JSON_VALUE)
+        .body("{\"surname\":\"smith\"}")
+        .`when`()["/search"]
+        .then()
+        .statusCode(200)
+        .extract()
+        .body()
+        .`as`(Array<OffenderDetail>::class.java)
+    Assertions.assertThat(results).hasSize(2)
```

Comment:
this is quite annoying of intellij - adding back in the fully qualified instead of static import

Reply:
yeah, I'll add the imports back in